src/main/resources/earlierversions/detect9.ps1

```diff
@@ -232,11 +232,29 @@ function Get-DetectJar ($DetectFolder, $DetectSource, $DetectVersionKey, $Detect
 
     if ($DetectSource -eq "") {
         if ($DetectVersion -eq "") {
-            $DetectVersionUrl = "https://sig-repo.synopsys.com/api/storage/bds-integrations-release/com/synopsys/integration/synopsys-detect?properties=" + $DetectVersionKey
+            $detectVersionKeySplit = $DetectVersionKey.split("_") # Split the key provided, as an example key would be of form: DETECT_LATEST_9
+            $detectVersionChar = $detectVersionKeySplit[-1]
+            $detectVersionNumber = [int]$detectVersionChar
+
+            # If major version is 9 or less, than download from synopsys location or else blackduck
+            if($detectVersionNumber -le 9) {
+                $DetectVersionUrl = "https://sig-repo.synopsys.com/api/storage/bds-integrations-release/com/synopsys/integration/synopsys-detect?properties=" + $DetectVersionKey
+            } else {
+                $DetectVersionUrl = "https://sig-repo.synopsys.com/api/storage/bds-integrations-release/com/blackduck/integration/detect?properties=" + $DetectVersionKey
+            }
             $DetectSource = Receive-DetectSource -ProxyInfo $ProxyInfo -DetectVersionUrl $DetectVersionUrl -DetectVersionKey $DetectVersionKey
         }
         else {
-            $DetectSource = "https://sig-repo.synopsys.com/bds-integrations-release/com/synopsys/integration/synopsys-detect/" + $DetectVersion + "/synopsys-detect-" + $DetectVersion + ".jar"
+            $detectVersionSplit= $DetectVersion.split(".") # Split the key provided, as an example version would be of form: 8.11.2
+            $detectVersionChar = $detectVersionSplit[0] # Get the major version from the variable
+            $detectVersionNumber = [int]$detectVersionChar
+
+            # If major version is 9 or less, than download from synopsys location or else blackduck
+            if($detectVersionNumber -le 9) {
+                $DetectSource = "https://sig-repo.synopsys.com/bds-integrations-release/com/synopsys/integration/synopsys-detect/" + $DetectVersion + "/synopsys-detect-" + $DetectVersion + ".jar"
+            } else {
+                $DetectSource = "https://sig-repo.synopsys.com/bds-integrations-release/com/blackduck/integration/detect/" + $DetectVersion + "/detect-" + $DetectVersion + ".jar"
```

Comment:
same comment for the detect8 script but would we want to download from repo.blackduck in these else cases?